### PR TITLE
Alejandro.gonzalez/vulnerability unvalidated redirect spring

### PR DESCRIPTION
**Pull request by jandro996**
work in progress

### CODE REVIEW

File: dd-java-agent/instrumentation/spring-webmvc-3.1/src/main/java/datadog/trace/instrumentation/springweb/AbstractUrlBasedViewInstrumentation.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import datadog.trace.api.iast.InstrumentationBridge;
+import datadog.trace.api.iast.sink.UnvalidatedRedirectModule;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.springframework.web.servlet.view.RedirectView;
+
+@AutoService(Instrumenter.class)
+public class AbstractUrlBasedViewInstrumentation extends Instrumenter.Iast
+    implements Instrumenter.ForTypeHierarchy {
```
Comment (manuel-alvarez-alvarez) [inline dd-java-agent/instrumentation/spring-webmvc-3.1/src/main/java/datadog/trace/instrumentation/springweb/AbstractUrlBasedViewInstrumentation.java:19]: Why a hierarchy?, it's a single type (you are instrumenting the base class only)

File: dd-java-agent/instrumentation/spring-webmvc-3.1/src/main/java/datadog/trace/instrumentation/springweb/AbstractUrlBasedViewInstrumentation.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        AbstractUrlBasedViewInstrumentation.class.getName() + "$RedirectionAdvice");
+  }
+
+  public static class RedirectionAdvice {
+    @Advice.OnMethodExit(suppress = Throwable.class)
+    public static void redirection(
+        @Advice.This Object urlBasedView, @Advice.Argument(0) final String url) {
+      final UnvalidatedRedirectModule module = InstrumentationBridge.UNVALIDATED_REDIRECT;
+      if (module != null && url != null && urlBasedView instanceof RedirectView) {
+        module.onRedirect(url);
```
Comment (manuel-alvarez-alvarez) [inline dd-java-agent/instrumentation/spring-webmvc-3.1/src/main/java/datadog/trace/instrumentation/springweb/AbstractUrlBasedViewInstrumentation.java:51]: For `RedirectView` it's clearly an unvalidated redirect but what about other views? they can be considered as an internal forward to a JSP, a mustache template ...

Comment (jandro996) [reply dd-java-agent/instrumentation/spring-webmvc-3.1/src/main/java/datadog/trace/instrumentation/springweb/AbstractUrlBasedViewInstrumentation.java:51]: I will check all the `UrlBasedView` implementation to decide which ones could drive into a invalidated redirect vulnerability

File: dd-smoke-tests/springboot/src/test/groovy/datadog/smoketest/IastSpringBootRedirectSmokeTest.groovy [hunk 3]
```diff
@@ -73,4 +73,43 @@ class IastSpringBootRedirectSmokeTest extends AbstractIastServerSmokeTest {
...
+  def "unvalidated  redirect from RedirectView is present"() {
+    setup:
+    String url = "http://localhost:${httpPort}/unvalidated_redirect_from_redirect_view?param=redirected"
+    def request = new Request.Builder().url(url).get().build()
+
+    when:
+    client.newCall(request).execute()
+
+    then:
+    hasVulnerabilityInLogs { vul -> vul.type == 'UNVALIDATED_REDIRECT' && vul.location.method == 'unvalidatedRedirectFromRedirectView' }
```
Comment (manuel-alvarez-alvarez) [inline dd-smoke-tests/springboot/src/test/groovy/datadog/smoketest/IastSpringBootRedirectSmokeTest.groovy:86]: Can you try checking in the spans first?

Comment (jandro996) [reply dd-smoke-tests/springboot/src/test/groovy/datadog/smoketest/IastSpringBootRedirectSmokeTest.groovy:88]: After talk with @smola we decide to solve this problem later (https://datadoghq.atlassian.net/browse/APPSEC-9861?atlOrigin=eyJpIjoiYTMzNzEwYTk0MjY2NDNkZjlkZTNjZWNmNTg3NjM4YTkiLCJwIjoiaiJ9)

**Comment by jandro996** [comment]
This feature is covered by https://github.com/DataDog/dd-trace-java/pull/5308

**Pull request closed without merging**